ui/src/components/SetupRequired.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState, useEffect } from 'react'
+import Ledger from '@daml/ledger';
+import { Form, Button, Loader } from 'semantic-ui-react';
+
+import { Operator } from '@daml.js/da-marketplace/lib/Marketplace/Operator';
+import { httpBaseUrl, deploymentMode, DeploymentMode } from "../config";
+import deployTrigger, { TRIGGER_HASH, MarketplaceTrigger, PublicAutomation, getPublicAutomation } from '../automation';
+import { Tile, logoHeader } from './common/OnboardingTile'
+import { useDablParties } from './common/common';
+
+const SetupRequired  = () => {
+  const [ deploying, setDeploying ] = useState(false);
+  const [ setupError, setSetupError ] = useState(false);
+  const [ adminJwt, setAdminJwt ] = useState('');
+  const { parties } = useDablParties();
+  const [ automations, setAutomations ] = useState<PublicAutomation[] | undefined>([]);
+  const publicParty = useDablParties().parties.publicParty;
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      getPublicAutomation(publicParty).then(autos => {
+        setAutomations(autos);
+        if (!!automations && automations.length > 0) {
+          clearInterval(timer);
+        }
+      });
+    }, 3000);
+    return () => clearInterval(timer);
+  }, [publicParty]);
```

Comment:
Mind explaining what this is doing? The timer inside the hook? I dont totally follow

Reply:
sure - I wanted the screen to change once the automation was made deployable, so this hits the deployable endpoint every 3 seconds until it returns with automations. If you know a better way of doing this, please let me know!